pom.xml

```diff
@@ -299,6 +299,53 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <version>3.1.0</version>
+        <configuration>
+          <createSourcesJar>true</createSourcesJar>
+          <shadeSourcesContent>true</shadeSourcesContent>
+          <relocations>
+            <relocation>
+              <pattern>com.google.common</pattern>
+              <shadedPattern>com.signalfx.shaded.google.common</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>com.google.thirdparty</pattern>
+              <shadedPattern>com.signalfx.shaded.google.thirdparty</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>com.fasterxml</pattern>
+              <shadedPattern>com.signalfx.shaded.fasterxml</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>org.apache</pattern>
+              <shadedPattern>com.signalfx.shaded.apache</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>com.yammer</pattern>
+              <shadedPattern>com.signalfx.shaded.yammer</shadedPattern>
+            </relocation>
+            <relocation>
+              <pattern>com.codahale</pattern>
+              <shadedPattern>com.signalfx.shaded.codahale</shadedPattern>
```

Comment:
I'm not sure you can shade codahale and yammer, as those modules appear to have user facing API surface area with types in those modules.

Simplified example from your tests:

```
MetricRegistry metricRegistery = new MetricRegistry();
        SignalFxReporter reporter = new SignalFxReporter.Builder(metricRegistery, new StaticAuthToken(""), "myserver").build();
```

The user needs to be able to interact with `SignalFxReporter.Builder` by passing a `MetricRegistry` instance they already have, not the shaded type.

I may not be understanding the purpose of these modules though. If so, ignore me :)

Reply:
The reason I called out Guava and Jackson for shading initially is (1) they are internal implementation details and (2) likely to cause downstream pain.

Reply:
Good points, I don't think I should be shading those. All other "internal" dependencies should still be shaded though. I'll send an updated diff.